src/Views/KeyboardLayoutView.vala

```diff
@@ -142,6 +149,24 @@ public class KeyboardLayoutView : AbstractInstallerView {
         }
 
         show_all ();
+
+        Idle.add (() => {
```

Comment:
It looks like this Idle clause is getting really big for something that is a lambda function, I suggest putting the contents of that to another function that will get called by `Idle.add`.